index.d.ts

```diff
@@ -0,0 +1,21 @@
+// Definitions by: ricardo-devis-agullo <https://github.com/ricardo-devis-agullo>
+
+import fastify = require("fastify");
+import ioredis = require("ioredis");
+
+import { Server, IncomingMessage, ServerResponse } from "http";
+
+declare module "fastify" {
+  interface FastifyInstance<HttpServer, HttpRequest, HttpResponse> {
+    redis: ioredis.Redis;
+  }
+}
+
+declare const fastifyBasicAuth: fastify.Plugin<
+  Server,
+  IncomingMessage,
+  ServerResponse,
```

Comment:
Would you mind using https://github.com/fastify/fastify-swagger/blob/2f7d7f68f8d87f16904b5da8bec38784b64ac5c8/index.d.ts#L70-L73 so we support http2 as well?

Reply:
hmm this one seems it breaks the tests as TS can´t find a viable type that its a union of both http and http2 servers...

Reply:
Check out fastify-swagger, it supports both... so it should be doable.

Reply:
I just checked and I'm not sure they are correctly set. I just changed something on the tests to an invalid type, and the typescript compiler passed!

<img width="559" alt="Typechecker not working" src="https://user-images.githubusercontent.com/16902309/67159537-38efa300-f346-11e9-8ec1-2b6787bc1b11.png">